### PR TITLE
Fix dup mounter cache behavior

### PR DESCRIPTION
**Pull request by rajyan**
dup should not prevent uploading

fixes https://github.com/carrierwaveuploader/carrierwave/issues/2700

### CODE REVIEW

**Comment by rajyan** [comment]
I'm very confused with the results, but these test fails inconsistently...

**Comment by rajyan** [comment]
I'm not sure yet with
* If this test is expressing the problem in the issue
* Why the tests are failing randomly

Also, testing locally with counter_culture, I still could not upload an image.

**Comment by rajyan** [comment]
> I'm very confused with the results, but these test fails inconsistently...

This was simply cause by test fixture setup...
The first commit passes, because 'test.jpg' is uploaded by another test and the uploaded files isn't cleaned,

I would fix up them.

**Comment by rajyan** [comment]
All problems are resolved.
I believe this pull request now fixes https://github.com/carrierwaveuploader/carrierwave/issues/2700

**Comment by rajyan** [comment]
@mshibuya 

The last commit in
https://github.com/carrierwaveuploader/carrierwave/pull/2690
broke the `@_mounters` cache, because the `dup` is called after `nil` is set.
Sorry that I couldn't catch it.

File: spec/orm/activerecord_spec.rb [hunk 1]
```diff
@@ -1956,6 +1960,15 @@ def reload(*)
       expect { @event.dup }.not_to change { @event[:image] }
     end
 
+    it "can upload a file" do
```
Comment (mshibuya) [inline spec/orm/activerecord_spec.rb:1963]:
```suggestion
    it "allows the original object to store a file" do
```

Can the title be a bit more descriptive like this?

**Comment by mshibuya** [comment]
Looks great except for the tiny issue I commented!
This is my mistake as I didn't understand the invocation order of `CarrierWave::Mount::Extension#initialize_dup` and `CarrierWave::ActiveRecord` 's `#initialize_dup` correctly... 😓

File: spec/orm/activerecord_spec.rb (hunk 1 above)
Comment (rajyan) [reply spec/orm/activerecord_spec.rb]: Thanks for your suggestion!

**Comment by mshibuya** [comment]
Thank you so much!

**Pull request merged**